src/sql-parser/src/parser.rs

```diff
@@ -1891,19 +1891,31 @@ impl<'a> Parser<'a> {
             Keyword::Kafka => {
                 self.expect_keyword(BROKER)?;
                 let broker = self.parse_literal_string()?;
-                let with_options = self.parse_opt_with_options()?;
-                CreateConnector::Kafka {
-                    broker,
-                    with_options,
-                }
+                self.expect_keyword(SECURITY)?;
+                let security = self.parse_kafka_security()?;
+                CreateConnector::Kafka { broker, security }
             }
             Keyword::Confluent => {
                 self.expect_keywords(&[SCHEMA, REGISTRY])?;
                 let registry = self.parse_literal_string()?;
-                let with_options = self.parse_opt_with_options()?;
+                let (mut username, mut password) = (None, None);
+                while let Ok(keyword) = self.expect_one_of_keywords(&[USERNAME, PASSWORD]) {
```

Comment:
This is susceptible to pathological input, i.e. `USERNAME=good USERNAME=bad` will only retain bad. We don't want to perform semantic analysis on the input here, so I think maybe doing something like collecting the keys and values--maybe using a `Vec<WithOption>`, which we ensure contains no duplicates at some other point.

Reply:
The goal had been to allow them to be specified in either order, but using `WithOption` defeats the main motivation of these changes with is to eliminate them in favor of defined syntax. I would rather just require the order and eliminate the `WITH` option if that is the necessary tradeoff.